src/libsyntax/error_codes.rs

```diff
@@ -363,6 +363,35 @@ and likely to change in the future.
 
 "##,
 
+E0704: r##"
+This error indicates that a incorrect visibility restriction was specified.
+
+Example of erroneous code:
+
+```compile_fail,E0704
+mod foo {
+    pub(foo) struct Bar {
+        x: i32
+    }
+}
+```
+
+To make struct `Bar` only visible in module `foo` the `in` keyword should be
+used:
+```
+mod foo {
+    pub(in crate::foo) struct Bar {
+        x: i32
+    }
+}
+#fn main() {}
```

Comment:
```suggestion
# fn main() {}
```

Reply:
Once you've done that change the test should pass correctly. r=me after fixing and squashing

r? @estebank